engine/src/main/java/org/pentaho/di/trans/streaming/common/BaseStreamStepMeta.java

```diff
@@ -148,6 +148,15 @@ public void check( List<CheckResultInterface> remarks, TransMeta transMeta,
     }
   }
 
+  @Override
+  public String getFileName() {
+    String returnString = this.fileName;
+    if ( ( null == returnString ) || returnString.isEmpty() ) {
```

Comment:
We've got Guava available to us, so you can use Strings.isNullOrEmpty().   Also, I'd prefer a ternary operator, but that's just personal style.  
return Strings.isNullOrEmpty(fileName) ? getTransformationPath() : fileName